### PR TITLE
Update DG: Brief implementation description of Gradebook

### LINKED ISSUE #123: Update Developer Guide

### PR DESCRIPTION
**Pull request by yutingzou**
Update DG: Brief implementation description of Gradebook

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -142,6 +142,20 @@ Classes used by multiple components are in the `seedu.TutorTracker.commons` pack
...
+####Proposed Implementation
+The proposed gradebook machanism is to facilitate the user to keep track of his/her
+own grades of different subjects for reference, which are internally stored as `gradeList`. Additionally,
+it implements the following operations:  
+* `Add a subject grade` - Add a subject grade to user's gradebook
+* `Delete a subject grade` - Delete a subject grade by subject name
+* `Edit a subject grade` - Edit a subject grade by subject name
+   
+These operations are exposed in the `Logic` interface by parsing respective `AddGradeCommand`, 
+`DeleteGradeCommand` and `EditGradeCommand`, as well as the `Model` interface as `Model#gradeList`.
```
Comment (Winniehyx) [inline docs/DeveloperGuide.md]: is this a typo?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -142,6 +142,20 @@ Classes used by multiple components are in the `seedu.TutorTracker.commons` pack
 
 ## **Implementation**
 
+### [Proposed] Gradebook Feature
+####Proposed Implementation
+The proposed gradebook machanism is to facilitate the user to keep track of his/her
```
Comment (Winniehyx) [inline docs/DeveloperGuide.md]: mechanism*

Comment (Winniehyx) [reply docs/DeveloperGuide.md]: should it be feature instead of mechanism?

**Review by Winniehyx — APPROVED** [review]
lgtm

**Review by glatiuden — APPROVED** [review]
LGTM

**Pull request merged**